packages/spor-react/src/theme/utils/border-utils.ts

```diff
@@ -0,0 +1,59 @@
+import { mode, StyleFunctionProps } from "@chakra-ui/theme-tools";
+import { getBoxShadowString } from "./box-shadow-utils";
+
+type State =
+  | "default"
+  | "hover"
+  | "active"
+  | "focus"
+  | "selected"
+  | "invalid"
+  | "disabled";
```

Comment:
Could this be pulled out into a shared object? And then we can just `Pick<State, 'the' | 'ones' | 'we' | 'need'>`?